### PR TITLE
[SwiftUI] Introduce SwiftBrowser

### PR DESCRIPTION
**Pull request by rr-codes**
#### 4f8ffdf6215072151355419da90a66dbd6384110
<pre>
[SwiftUI] Introduce SwiftBrowser
<a href="https://bugs.webkit.org/show_bug.cgi?id=284239">https://bugs.webkit.org/show_bug.cgi?id=284239</a>
<a href="https://rdar.apple.com/135306861">rdar://135306861</a>

Reviewed by Aditya Keerthi.

Introduce a project similar to MiniBrowser using the modern SwiftUI implementation.

Current feature list:

* Load file and website URLs
* View the title of the current page
* Navigation progress bar
* Ability to stop loading or reload, either via the main UI or a Menu Bar command
* Back/forward buttons, and the ability to view the back/forward list when long pressing on the buttons
* Keyboard shortcut support for reload, back, and forward
* Multiple window and tab support
* Universal app
* Export as PDF option
* Ability to import file URLs
* A Settings view, which currently supports setting the home page

* Tools/SwiftBrowser/Configurations/Base.xcconfig: Added.
* Tools/SwiftBrowser/Configurations/DebugRelease.xcconfig: Added.
* Tools/SwiftBrowser/Configurations/SwiftBrowser.xcconfig: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/AccentColor.colorset/Contents.json: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/Artwork - Light.png: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/Contents.json: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/icon_128x128.png: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/icon_128x128@2x 1.png: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/icon_128x128@2x.png: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/icon_16x16.png: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/icon_16x16@2x 1.png: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/icon_16x16@2x.png: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/icon_256x256@2x.png: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/icon_32x32@2x.png: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/icon_512x512.png: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/icon_512x512@2x.png: Added.
* Tools/SwiftBrowser/Resources/Assets.xcassets/Contents.json: Added.
* Tools/SwiftBrowser/Resources/SwiftBrowser.entitlements: Added.
* Tools/SwiftBrowser/Resources/iOS/Info.plist: Added.
* Tools/SwiftBrowser/Resources/mac/Info.plist: Added.

Configurations and resources.

* Tools/SwiftBrowser/Source/BrowserView.swift: Renamed from Tools/MiniBrowser/SwiftUI/MiniBrowserView.swift.
* Tools/SwiftBrowser/Source/BrowserViewModel.swift: Renamed from Tools/MiniBrowser/SwiftUI/BrowserViewModel.swift.
(PDF.transferRepresentation):
(BrowserViewModel.exportedPDF):
(BrowserViewModel.openURL(_:)):
(BrowserViewModel.didReceiveNavigationEvent(_:)):
(BrowserViewModel.navigateToSubmittedURL):
(BrowserViewModel.exportAsPDF):
(BrowserViewModel.didExportPDF(_:any:)):
* Tools/SwiftBrowser/Source/ContentView.swift: Added.
(body):
(ContentView.body):
* Tools/SwiftBrowser/Source/SwiftBrowser.swift: Added.
(SwiftBrowserApp.body):

The actual implementation.

* Tools/SwiftBrowser/SwiftBrowser.xcodeproj/project.pbxproj: Added.
* Tools/SwiftBrowser/SwiftBrowser.xcodeproj/xcshareddata/xcschemes/SwiftUIMiniBrowser.xcscheme: Added.

Canonical link: <a href="https://commits.webkit.org/287572@main">https://commits.webkit.org/287572@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/47a663d2ccdf965407964c9dc4fefa6ebb6ddf93

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/80120 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/59122 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/33533 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/84637 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/31096 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/82231 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/68183 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/7420 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/62639 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/20461 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/83189 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/52713 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/72974 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/42943 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/50041 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/27128 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/29556 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/71169 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/27635 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/86069 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/7340 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/5197 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/70917 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/7515 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/68813 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/70156 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/14144 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/13096 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12396 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/7304 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/7143 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/10663 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/8948 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/UIProcess/API/Swift/WebPage.swift [hunk 1]
```diff
@@ -146,6 +146,7 @@ public class WebPage_v0 {
     lazy var backingWebView: WKWebView = {
         let webView = WKWebView(frame: .zero, configuration: WKWebViewConfiguration(configuration))
         webView.navigationDelegate = backingNavigationDelegate
+        webView.scrollView.bouncesVertically = false
```
Comment (rr-codes) [inline Source/WebKit/UIProcess/API/Swift/WebPage.swift]: remove

File: Tools/SwiftBrowser/Configurations/DebugRelease.xcconfig [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2010, 2013 Apple Inc. All rights reserved.
```
Comment (rr-codes) [inline Tools/SwiftBrowser/Configurations/DebugRelease.xcconfig]: update

File: Tools/SwiftBrowser/Configurations/SwiftBrowser.xcconfig [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2010 Apple Inc. All rights reserved.
```
Comment (rr-codes) [inline Tools/SwiftBrowser/Configurations/SwiftBrowser.xcconfig]: update

File: Tools/SwiftBrowser/SwiftBrowser.xcodeproj/xcshareddata/xcschemes/SwiftUIMiniBrowser.xcscheme [hunk 4]
```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rr-codes) [inline Tools/SwiftBrowser/SwiftBrowser.xcodeproj/xcshareddata/xcschemes/SwiftUIMiniBrowser.xcscheme]: is this file actually needed?

File: Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/Artwork - Light.png
Comment (rr-codes) [inline Tools/SwiftBrowser/Resources/Assets.xcassets/AppIcon.appiconset/Artwork - Light.png:1]: a slightly better-than-nothing icon

**Review by pxlcoder — COMMENTED** [review]
Maybe the `Source/WebKit/UIProcess/API/` changes should be split out into a separate change that isn't about introducing "SwiftBrowser"?

**Review by pxlcoder — APPROVED** [review]
